### ISSUE TITLE
altium2kicad creates invalid .kicad_pcb file

### ISSUE THREAD

**Issue by BreakingSiam** [thesourcerer8/altium2kicad#127]
After running altium2kicad on a large Altium project, it creates a .kicad_pcb file which Kicad version "6.0.4-6f826c9f35~116~ubuntu18.04.1, release build" reports "does not appear to be a valid KiCad project file". 

The generated .kicad_pcb file is 13891785 bytes long, and on visual inspection of the beginning of the file appears to be a valid KiCad project file.

During conversion of the altium project, after displaying writing to "Regions6/Data.dat.txt", altium2kicad reported:

Overflow!
Overflow!
Overflow!
Overflow!
Overflow!
Overflow!
Overflow!
Overflow!
Overflow!
Error: substring out of range: length:509 pos:18799 wantedsize:2 field:len[2] (main,../convertpcb.pl:2901 - main::msubstr)
Use of uninitialized value in unpack at ../convertpcb.pl line 2901.
...

The complete conversion output is attached.

[oo.txt](https://github.com/thesourcerer8/altium2kicad/files/9029142/oo.txt)

**Comment by thesourcerer8** [comment]
Please provide me with your altium file, so that I can analyze the problem. I couldn't reach you through email.

**Comment by thesourcerer8** [comment]
I just tried the new Altium Importer that has been integrated into Kicad itself (one in PcbNew and on in EESchema, to be found in "File -> Import -> Non-KiCad Files", and the results are much better than altium2kicad, so I would suggest to you use the KiCad integrated importer yourself instead of altium2kicad. If it works for you, please close this issue. If it doesn't work for you, please let me know.

**Comment by bjornfor** [comment]
Has the integrated imprter in KiCAD superseded this project?

**Comment by thesourcerer8** [comment]
Yes, it was developed based on this converter, and it seems to have surpassed the quality of this converter now. So unless anyone still needs this converter, I would declare it superseded now.

**Comment by dzid26** [comment]
> Yes, it was developed based on this converter, and it seems to have surpassed the quality of this converter now. So unless anyone still needs this converter, I would declare it superseded now.

It works better indeed. Although, for reference, it is in nightly builds. 
And for me, it only worked after first exporting to Circuit Maker / Studio. 
https://forum.kicad.info/t/trouble-importing-pcb-files-from-solidworks-pcb-altium/41689/10?u=dzid26
